src/type/definition.js

```diff
@@ -506,6 +506,11 @@ function defineFieldMap<TSource, TContext>(
       `${type.name}.${fieldName} field type must be Output Type but ` +
       `got: ${String(field.type)}.`
     );
+    invariant(
+      isValidResolver(field.resolve),
```

Comment:
you could probably just inline this as `!field.resolve || typeof field.resolve === 'function'`